### PR TITLE
Add label selectors for monitors and rules

### PR DESCRIPTION
**Pull request by glrf**
We introduce a opt-in and opt-out labeling system for prometheusrules and monitors.

Pod and Service Monitors are opt-out. By setting `monitoring.syn.tools/enabled: "false"` we can make Prometheus ignore a monitor, even if the namespace it labeled to be monitored.

PrometheusRules are opt-in. Prometheus will only pick up rules if they are in a monitored (i.e. labeled) namespace AND the rule is labeled with `monitoring.syn.tools/enabled: "true"`. This should ensure that rules are added consciously and prevent a massive import of upstream alerts that are not actionable or don't meet our standards.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/cluster-monitoring.adoc [hunk 1]
```diff
@@ -8,7 +8,12 @@ However it also supports picking up metrics and alerts from other workloads depl
...
 
-This sets namespace slectors on every Prometheus instance that will result in them picking up all `ServiceMonitors`, `PodMonitors`, `Probes`, and `PrometheusRules` in namespaces with the label `monitoring.syn.tools/<instance>`.
+This sets namespace selectors on every Prometheus instance that will result in them picking up all `ServiceMonitors`, `PodMonitors`, and `Probes` in namespaces with the label `monitoring.syn.tools/<instance>`.
+
+`PrometheusRules` are only pick up rules if they're in a labeled namespace AND the rule is labeled with `monitoring.syn.tools/enabled: "true"`.
+This should ensure that rules are added consciously and prevent a massive import of upstream alerts that aren't actionable or don't meet our standards.
+
+You also have the option to disable `ServiceMonitors`, `PodMonitors`, or `Probes` by labeling them with `monitoring.syn.tools/enabled: "false"`.
 
 The example blow will make the Prometheus instance `default-instance` pick up all namespaces with a label `monitoring.syn.tools/default-instance: "true"`.
```
Comment (simu) [inline docs/modules/ROOT/pages/how-tos/cluster-monitoring.adoc:18]: This is not strictly related to this PR, I just noticed the typo during the review.

```suggestion
The example below will make the Prometheus instance `default-instance` pick up all namespaces with a label `monitoring.syn.tools/default-instance: "true"`.
```

**Review by simu — APPROVED** [review]
LGTM

**Review by bastjan — APPROVED** [review]
